dask/dataframe/io/parquet/arrow.py

```diff
@@ -317,6 +318,36 @@ def _need_fragments(filters, partition_keys):
     return bool(filtered_cols - partition_cols)
 
 
+def _hive_dirname(name, val):
+    # Simple utility to produce hive directory name
+    val = "__HIVE_DEFAULT_PARTITION__" if pd.isna(val) else val
+    return f"{name}={val}"
+
+
+def _process_kwargs(partitioning=None, warn=False, **kwargs):
```

Comment:
I see us calling this function in three different places. Is it possible to just call it once early in the call stack and have it automatically passed down to where it's used?

Reply:
We are already calling this in the minimum number of places: Once in `_collect_dataset_info` (which will always be executed on the client), once in `_collect_file_parts` (which may need to execute on a worker when `read_parquet` is called), and once in `_read_table` (which may execute on a worker at compute time).